### PR TITLE
refactor move cloudfoundry to his own module

### PR DESCRIPTION
**Pull request by matedeilo**
## Summary
Move cloudfoundry to his own module azure-spring-boot-cloudfoundry

## Issue Type
- New Feature

## Starter Names
Add new starter
  - azure cloudfoundry spring boot starter

## Additional Information
Co-authored-by: Eddu Melendez <eddu.melendez@gmail.com>

### CODE REVIEW

File: azure-spring-boot-starters/azure-cloudfoundry-spring-boot-starter/pom.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    <artifactId>azure-cloudfoundry-spring-boot-starter</artifactId>
+    <packaging>jar</packaging>
+
+    <name>Azure Cloud Foundry Spring Boot Starter Sample</name>
+
+    <properties>
+        <project.rootdir>${project.basedir}/../..</project.rootdir>
+    </properties>
+
+    <dependencies>
```
Comment (eddumelendez) [inline azure-spring-boot-starters/azure-cloudfoundry-spring-boot-starter/pom.xml]: add this dependency too

```
<dependency>
    <groupId>com.microsoft.azure</groupId>
    <artifactId>azure-spring-boot-starter</artifactId>
</dependency>
```

File: azure-spring-boot-cloudfoundry/pom.xml [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        <groupId>com.microsoft.azure</groupId>
+        <artifactId>azure-spring-boot-parent</artifactId>
+        <version>2.1.8-SNAPSHOT</version>
+        <relativePath>../azure-spring-boot-parent/pom.xml</relativePath>
+    </parent>
+
+    <artifactId>azure-spring-boot-cloudfoundry</artifactId>
+    <packaging>jar</packaging>
+
+    <dependencies>
```
Comment (eddumelendez) [inline azure-spring-boot-cloudfoundry/pom.xml]: Let's review the dependencies. I think not of all them are needed in this module

File: azure-spring-boot-samples/azure-cloud-foundry-service-sample/pom.xml [hunk 3]
```diff
@@ -20,8 +20,7 @@
     <properties>
         <project.rootdir>${project.basedir}/../..</project.rootdir>
     </properties>
-
-
+
```
Comment (eddumelendez) [inline azure-spring-boot-samples/azure-cloud-foundry-service-sample/pom.xml]: remove this empty line

File: azure-spring-boot/src/main/resources/META-INF/spring.factories [hunk 4]
```diff
@@ -1,4 +1,3 @@
-org.springframework.boot.env.EnvironmentPostProcessor=com.microsoft.azure.spring.cloudfoundry.environment.VcapProcessor,\
```
Comment (eddumelendez) [inline azure-spring-boot/src/main/resources/META-INF/spring.factories]: do not remove `org.springframework.boot.env.EnvironmentPostProcessor`. It should looks like

```
org.springframework.boot.env.EnvironmentPostProcessor=\
com.microsoft.azure.keyvault.spring.KeyVaultEnvironmentPostProcessor,\
com.microsoft.azure.spring.autoconfigure.sqlserver.AlwaysEncryptedEnvironmentPostProcessor
```

**Comment by neuqlz** [comment]
Thanks for your contribution, we will review it later, then give you a response.

File: azure-spring-boot-starters/azure-cloudfoundry-spring-boot-starter/pom.xml (hunk 1 above)
Comment (matedeilo) [reply azure-spring-boot-starters/azure-cloudfoundry-spring-boot-starter/pom.xml]: Added it

File: azure-spring-boot-cloudfoundry/pom.xml (hunk 2 above)
Comment (matedeilo) [reply azure-spring-boot-cloudfoundry/pom.xml]: I've remove them and i left the required

File: azure-spring-boot-samples/azure-cloud-foundry-service-sample/pom.xml (hunk 3 above)
Comment (matedeilo) [reply azure-spring-boot-samples/azure-cloud-foundry-service-sample/pom.xml]: done

File: azure-spring-boot/src/main/resources/META-INF/spring.factories (hunk 4 above)
Comment (matedeilo) [reply azure-spring-boot/src/main/resources/META-INF/spring.factories]: Added it

**Comment by eddumelendez** [comment]
I worked with @matedeilo in order to solve this and we are still polishing. @neuqlz this PR is part of #712 we will wait for your comments.

File: azure-spring-boot-cloudfoundry/pom.xml (hunk 2 above)
Comment (eddumelendez) [reply azure-spring-boot-cloudfoundry/pom.xml]: I still see many dependencies may not needed here. I think `org.springframework.boot:spring-boot-autoconfigure` should be enough with the test library. Let's take a look into it again please.

**Comment by chenrujun** [comment]
I'll close this PR because it's not updated for long time.

**Pull request closed without merging**